src/sentry/incidents/models.py

```diff
@@ -657,6 +657,35 @@ class Meta:
         unique_together = (("alert_rule", "label"),)
 
 
+class AlertRuleActivationConditionType(Enum):
+    RELEASE_CREATION = 0
+    DEPLOY_CREATION = 1
+
+
+@region_silo_only_model
+class AlertRuleActivationCondition(Model):
+    """
+    This model represents the activation condition for an activated AlertRule
+
+    label is an optional identifier for an activation condition
+    condition_type is AlertRuleActivationConditionType
+    TODO: implement extra query params for advanced conditional rules (eg. +10m after event occurs)
+    """
+
+    __relocation_scope__ = RelocationScope.Organization
+
+    alert_rule = FlexibleForeignKey("sentry.AlertRule", related_name="activation_conditions")
+    label = models.TextField()
+    condition_type = models.SmallIntegerField(null=True)
+
+    date_added = models.DateTimeField(default=timezone.now)
+
+    class Meta:
+        app_label = "sentry"
+        db_table = "sentry_alertruleactivationcondition"
+        unique_together = (("alert_rule", "label"),)
```

Comment:
Just want to make sure that we definitely want this to be unique. So after the first of these, a label would have to be provided

Reply:
Yup that seems fine to me.

Currently we would never expect more than 1 of these to exist per alert rule - however, we _could_ potentially have a future where we can introduce multiple conditions that might trigger an activated alert rule 👍

Reply:
Would we be better served by a `type` column that's backed by an enum in that case?

Reply:
Yea i considered that - however we have a use case where we'll want to introduce more complex conditions as well like "10m _after_ a deploy has been created", or "after a deployed change has hit 100 events" or something.

I figured a separate table for this would be the best option and we can define how those extra conditions can look later